signature/jws/signer.go

```diff
@@ -105,6 +160,27 @@ func jwtToken(alg string, claims jwt.Claims) *jwt.Token {
 	}
 }
 
+func keySpecFromJWS(alg string) (notation.KeySpec, error) {
+	var keySpec notation.KeySpec
+	switch alg {
+	case jwt.SigningMethodES256.Name:
+		keySpec = notation.EC_256
```

Comment:
Would it be better to create a `keySpecFromKey(key crypto.PrivateKey)` method instead, currently we get a JWS alg from privatekey, and then keyspec from alg.

Reply:
`NewSignerWithCertificateChain` allows to pass custom keys other than `rsa.PrivateKey` and `ecdsa.PrivateKey`. If we derive the keySpec from the key instead of the algorithm, we will be forcing callers to pass one of these two concrete types, as we the only way to implement your suggestion is to use a type switch for known types, similar to how `SigningMethodFromKey` is implemented.

Reply:
Hmm, if we don't allow custom signers, which is the value of `NewSignerWithCertificateChain`? AFAIU `NewSigner` covers all use cases.

Anyway, I'll implement your suggestion.

Reply:
I think you are right `NewSignerWithCertificateChain` need not be a public method.

Reply:
If we unexport `NewSignerWithCertificateChain` then exporting `SigningMethodFromKey` neither makes sense, as it only makes sense in conjunction with `NewSignerWithCertificateChain`.

Anyway, these changes go out of the scope of this PR.

Reply:
At the end I've removed `NewSignerWithCertificateChain` and unexported `SigningMethodFromKey`.